internal/pgdb/pgdb_index.go

```diff
@@ -175,5 +175,94 @@ func getCommonIndexes(ctx pgsgo.Context, m pgs.Message) ([]*indexContext, error)
 		},
 	}
 
-	return []*indexContext{primaryIndex, pkskIndexBroken, pkskIndex, ftsIndex}, nil
+	// again loop through and look for vector / hnsw indexes
+
+	rv := []*indexContext{primaryIndex, pkskIndexBroken, pkskIndex, ftsIndex}
+
+	// iterate message for vector behavior options
+	for _, field := range m.Fields() {
+		ext := pgdb_v1.FieldOptions{}
+		_, err := field.Extension(pgdb_v1.E_Options, &ext)
+		if err != nil {
+			return nil, fmt.Errorf("pgdb: getField: failed to extract Message extension from '%s': %w", field.FullyQualifiedName(), err)
+		}
+		if ext.MessageBehavoir != pgdb_v1.FieldOptions_MESSAGE_BEHAVOIR_VECTOR {
+			continue
+		}
+		if !field.Type().IsRepeated() {
```

Comment:
this can be a shared function `assertFieldIsVectorShape()`?

Reply:
https://github.com/ductone/protoc-gen-pgdb/pull/66/commits/d311d794661bff07d3d381d7843587d3e2a70fb2